### PR TITLE
[vm] Remove VMConfig

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation
This is a followup cleanup for #2932. `VMConfig` is removed from `libra_config` crate as all VM related config should be consensus critical and should thus live on chain.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

`cargo test`

## Related PRs
#2932

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: vm-validator/src/vm_validator.rs [hunk 1]
```diff
@@ -34,11 +34,11 @@ pub struct VMValidator {
 
 impl VMValidator {
     pub fn new(
```
Comment (sblackshear) [inline vm-validator/src/vm_validator.rs]: Delete `config` parameter + refactor clients of `VMValidator::new`?

Comment (runtian-zhou) [reply vm-validator/src/vm_validator.rs]: I was a bit hesitated to do that because VMValidator might want to store other configs in the future.

Comment (sblackshear) [reply vm-validator/src/vm_validator.rs]: Let's clean up unused params now and let someone add them back if they're needed in the future?

**Pull request closed without merging**